docs/Versioning.md

```diff
@@ -0,0 +1,73 @@
+# ML-Agents Versioning
+
+## Context
+As the ML-Agents project evolves into a more mature product, we want to communicate the process
+we use to version our packages and the data that flows into, through, and out of them clearly.
+Our project now has four packages (1 Unity, 3 Python) along with artifacts that are produced as
+well as consumed.  This document covers the versioning for these packages and artifacts.
+
+## Packages
+All of the software packages, and their generated artifacts will be versioned.  Any automation
+tools will not be versioned.
+
+### Unity package
+Package name: com.unity.ml-agents
+- Versioned following [Semantic Versioning Guidelines](https://www.semver.org)
+- This package produces one artifact, the `.demo` files.  These files will have integer
+    versioning. This means their version will increment by 1 at each change.  The
+    com.unity.ml-agents package must be backward compatible with version changes
+    that occur between minor versions.  For example, consider that com.unity.ml-agents
+    is at version 1.0.0 and the demo files are at version one.  If the demo files
+    change to version 2, the next release of com.unity.ml-agents at version 1.1.0
+    shall be able to read both of these formats.  If the demo files were to change to
+    version 3 and com.unity.ml-agents to version 2.0.0, support for demo versions 1 and
+    2 could be dropped for com.unity.ml-agents version 2.0.0.
+- This package consumes an artifact of the training process: the `.nn` file.  The
+    com.unity.ml-agents package will need to support the version of `.nn` files
+    which existed at its 1.0.0 release.
```

Comment:
"... for the remainder of the 1.0.0 release cycle"? This makes it sounds like we'll support the current format indefinitely.

Reply:
change the wording a bit, let me know if it's better.